nautobot/core/apps/__init__.py

```diff
@@ -109,6 +114,80 @@ def register_menu_items(tab_list):
+    define different parts of the layout.
+
+    These objects are converted into a dictionary to be stored inside of the Nautobot registry.
+    """
+    name, app_name = app_name.split(".")
+    template_path = f"{name}/{app_name}/templates/{app_name}/inc/"
+    registry_panels = registry["homepage_layout"]["panels"]
+    for panel in homepage_layout:
+        panel_perms = set()
+        panel.template_path = template_path
+        if isinstance(panel, HomePagePanel):
+            create_or_check_entry(registry_panels, panel, panel.name, f"{panel.name}")
+            registry_items = registry_panels[panel.name]["items"]
+            if panel.custom_template:
+                registry["homepage_layout"]["total_items"] += 1
+
+            for item in panel.items:
+                if isinstance(item, HomePageItem):
+                    item.template_path = template_path
+                    create_or_check_entry(registry_items, item, item.name, f"{panel.name} -> {item.name}")
+                    registry["homepage_layout"]["total_items"] += 1
+                    panel_perms |= set(perms for perms in item.permissions)
+                elif isinstance(item, HomePageGroup):
+                    item.template_path = template_path
+                    create_or_check_entry(registry_items, item, item.name, f"{panel.name} -> {item.name}")
+                    for group_item in item.items:
+                        if isinstance(group_item, HomePageItem):
+                            create_or_check_entry(
+                                registry_items[item.name]["items"],
+                                group_item,
+                                group_item.name,
+                                f"{panel.name} -> {item.name} -> {group_item.name}",
+                            )
+                            registry["homepage_layout"]["total_items"] += 1
+                    panel_perms |= set(perms for perms in group_item.permissions)
+                    registry_items[item.name]["items"] = OrderedDict(
+                        sorted(registry_items[item.name]["items"].items(), key=lambda kv_pair: kv_pair[1]["weight"])
+                    )
+                else:
+                    raise TypeError(
+                        f"Second level objects need to be an instance of HomePageGroup or HomePageItem: {item}"
+                    )
+
+            registry_panels[panel.name]["items"] = OrderedDict(
+                sorted(registry_items.items(), key=lambda kv_pair: kv_pair[1]["weight"])
+            )
+        else:
+            raise TypeError(f"Top level objects need to be an instance of HomePagePanel: {panel}")
+        registry_panels[panel.name]["permissions"] = panel_perms
+
+    registry["homepage_layout"]["panels"] = OrderedDict(
+        sorted(registry_panels.items(), key=lambda kv_pair: kv_pair[1]["weight"])
+    )
+    registry["homepage_layout"]["items_per_column"] = registry["homepage_layout"]["total_items"] / 3
```

Comment:
Instead of programmatically dividing the panels amongst N columns, it *should* be possible to use CSS to define all of the panels into a single container and let the browser dynamically re-flow it into a 3-column layout. This might be better as it doesn't rely on any hard-coded assumptions about the height of each individual panel being equal or such. 

See for example: https://css-tricks.com/guide-responsive-friendly-css-columns/

Reply:
That was easier to do than I thought! I'll do a PR against your branch.

Reply:
https://github.com/carbonarok/nautobot/pull/1